### PR TITLE
fix(core): address Avatar Group issues

### PR DESCRIPTION
**Pull request by khotcholava**
fix(core): Resolve focus retention & rendering issues, add feature enhancements

Closes [#12369](https://github.com/SAP/fundamental-ngx/issues/12369)

## Description
- Fixed focus bug where focus remained on the first avatar group control after interacting with the second one.
- Fixed rendering issue where only "+n more" button displayed after switching tabs instead of the full avatar group row.
- Added feature to open group popover from the center of the avatar group.
- Added configuration option to make "+n more" button circular to match avatars typically being in circle shapes.
- Introduced property to set the number of visible avatars before displaying the "+n more" button.

### CODE REVIEW

File: libs/docs/core/avatar-group/avatar-group-docs.component.html [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    <p>
-        In this example, which is called <code>individual</code> type avatar group, which means that every item is
-        displayed separately, every item can be focused separately and details template will be displayed for that item.
-    </p>
-</description>
-<component-example>
-    <fd-avatar-group-default-example></fd-avatar-group-default-example>
-</component-example>
-<code-example [exampleFiles]="avatarGroupDefaultExample"></code-example>
+<!--<fd-docs-section-title id="default" componentName="AvatarGroup"> Avatar Group Standard Usage </fd-docs-section-title>-->
```
Comment (mikerodonnell89) [inline libs/docs/core/avatar-group/avatar-group-docs.component.html:1]: Accidental commit?

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Looks good but what is going on with all of the commented out html?

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
In the "group type standard usage" examples, after opening the popover, i can not navigate through the avatars with the arrow keys. Because each avatar in the popover is clickable, we do need this functionality with keyboard control as well

**Comment by khotcholava** [comment]
@mikerodonnell89  In this [example](https://sap.github.io/ui5-webcomponents/components/AvatarGroup/#grouping), the web components avatar group features tab navigation but lacks arrow navigation.

I'm currently working on implementing tab navigation.

**Comment by khotcholava** [comment]
@mikerodonnell89 Implemented arrow navigation.

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
- For the "Group Type" examples, I don't think each avatar should be focusable/tabbable. It's just supposed to be the one focus around the whole group. The user shouldn't be able to do this:
https://github.com/user-attachments/assets/b5cc768f-c88f-4164-9fa1-798331accd90

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
If I open the group type with the space bar, I'd expect the first avatar in the popover to receive focus. It doesn't, and also pressing tab still goes to focus each individual avatar in the non-popover part of the component:

https://github.com/user-attachments/assets/59721035-1f2e-4b9f-960b-5b4ed92020ff

File: libs/core/avatar-group/components/default-avatar-group-overflow-body/default-avatar-group-overflow-body.component.ts [hunk 2]
```diff
@@ -73,6 +74,16 @@ export class DefaultAvatarGroupOverflowBodyComponent implements AfterViewInit, O
     get isRtl(): boolean {
         return !!this._rtlService?.rtl.value;
     }
+
+    /**
+     * List of avatars to be rendered in the overflow popover.
+     **/
+    avatars = input<AvatarGroupItemRendererDirective[]>([]);
+
+    /**
```
Comment (dpavlenishvili) [inline libs/core/avatar-group/components/default-avatar-group-overflow-body/default-avatar-group-overflow-body.component.ts:83]: missing declaration or wrong place for this jsDoc.

**Comment by droshev** [comment]
@mikerodonnell89 @dpavlenishvili Can you review it again?

**Comment by droshev** [comment]
@dpavlenishvili any update?

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Does not look like the first issue (regarding the focus staying on the first avatar group after interacting with a second) was fixed. In this example I've used the mouse to click both, then the escape key to close the second group

https://github.com/user-attachments/assets/63ca5cf0-de58-4073-a94a-a4bf99f31221

**Comment by droshev** [comment]
@khotcholava Can you check the e2e?

File: libs/core/avatar-group/avatar-group.component.ts [hunk 3]
```diff
@@ -88,6 +94,10 @@ export class AvatarGroupComponent implements AvatarGroupHostConfig {
     @Input()
     size: AvatarGroupHostConfig['size'] = 'l';
 
+    /** Popover placement */
+    @Input()
+    popoverPlacement: Placement | null = null;
```
Comment (dpavlenishvili) [inline libs/core/avatar-group/avatar-group.component.ts:99]: use Nullable type.

File: libs/core/avatar-group/components/avatar-group-host.component.ts
Comment (dpavlenishvili) [inline libs/core/avatar-group/components/avatar-group-host.component.ts:1]: add or update relevant unit tests for maxVisibleItems

File: libs/core/popover/popover-service/popover.service.ts [hunk 4]
```diff
@@ -127,6 +131,13 @@ export class PopoverService extends BasePopoverClass {
                 this._removeOverlay(this._modalBodyClass, this._modalTriggerClass);
             }
         });
+
+        // Add keydown listener for space key
+        this._renderer.listen('document', 'keydown', (event: KeyboardEvent) => {
```
Comment (dpavlenishvili) [inline libs/core/popover/popover-service/popover.service.ts:136]: Listening for keydown events globally on the document level can lead some performance issues, if it is possible scope the event listener to the popover element directly. 
for example:
`
const popoverElement = this._elementRef.nativeElement;
this._renderer.listen(popoverElement, 'keydown', (event: KeyboardEvent) => {
    if (event.code === 'Space' && this.isOpen) {
        this._storeAndFocusFirstTabbable();
    }
});
`

File: libs/core/avatar-group/components/avatar-group-host.component.ts
Comment (khotcholava) [reply libs/core/avatar-group/components/avatar-group-host.component.ts:1]: There is not e2e test for avatar group host

File: libs/core/avatar-group/avatar-group.component.ts (hunk 3 above)
Comment (khotcholava) [reply libs/core/avatar-group/avatar-group.component.ts:99]: PopoverPlacement type is Placement | null. It's not Nullable<Placement>. If I change it I'm afraid it will broke already existing functionality

**Comment by droshev** [comment]
@khotcholava needs to update with the comments from UX, check e2e/unit and make it as PR. Needs to be reviewed afterwards

**Comment by droshev** [comment]
@khotcholava needs to update with the comments from UX, check e2e/unit and make it as PR. Needs to be reviewed afterwards

**Comment by droshev** [comment]
@khotcholava Can you summarize what you did and what needs to be implemented?

**Comment by mikerodonnell89** [comment]
Closing as this is quite dated - we will need to readdress #12369
